### PR TITLE
feat: add a config to control which window to focus on when closing window

### PR DESCRIPTION
**Pull request by ParaN3xus**
#### Describe your PR, what does it fix/add?

This PR introduces a new configuration option, `input::focus_on_close`, which controls the window focus behavior after a window is closed.

Given the existing focus-related settings within the `input` category, I have placed this configuration there as well.

`input::focus_on_close` is an integer, and currently supports two values:
- **0**: Maintains the current behavior, where focus shifts to the next window candidate as determined by `getNextWindowCandidate`.
- **1**: Changes focus to the window under the cursor, which I believe offers a more intuitive user experience.

The reason for using an integer instead of a boolean is to allow for future extensibility. For example, the option could be extended to focus on the previously focused window, or implement other behaviors, depending on user needs or future contributions.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

No known issues.

#### Is it ready for merging, or does it need work?

Yes, it is ready for merging.

### CODE REVIEW

**Comment by Visual-Dawg** [comment]
This is quite cool. I am not a maintainer nor know C++, but I think using a string like "cursor" and "next" instead of an integer is nicer to understand for the end-user :D

**Review by vaxerski — COMMENTED** [review]
clang-format and wiki mr needed :)

**Comment by vaxerski** [comment]
> This is quite cool. I am not a maintainer nor know C++, but I think using a string like "cursor" and "next" instead of an integer is nicer to understand for the end-user :D

pls dont

**Comment by Visual-Dawg** [comment]
Pf, strings are nicer

**Comment by ParaN3xus** [comment]
Apologies for missing the PR guidelines. I've now applied clang-format and updated the wiki([hyprland-wiki/pull/753](https://github.com/hyprwm/hyprland-wiki/pull/753)). Thank you for the feedback!

**Comment by ParaN3xus** [comment]
@Visual-Dawg 
I considered using strings, but chose integers to align with the existing configurations in the project. 

Settings like `follow_mouse`, `off_window_axis_events`, and `emulate_discrete_scroll`all use integers to represent different modes of operation. These settings are functionally similar in that they each control various behavioral outcomes.

I agree that strings can be more user-friendly, but in this case, I prioritized consistency with existing configurations.

**Comment by vaxerski** [comment]
strings will be parsable in the future once the descriptions branch gets merged (I should merge it)

**Comment by vaxerski** [comment]
ye I merged #7377 can you add your opt to configDescriptions.hpp too?

**Comment by ParaN3xus** [comment]
I've completed the rebase and updated `ConfigDescriptions.hpp`. Please review the latest changes.
@vaxerski

File: src/config/ConfigDescriptions.hpp
Comment (vaxerski) [inline src/config/ConfigDescriptions.hpp:460]: data type should be CHOICE

File: src/config/ConfigDescriptions.hpp
Comment (vaxerski) [inline src/config/ConfigDescriptions.hpp:458]: [0/1] redundant, choice will handle it

**Review by vaxerski — COMMENTED** [review]
rest ok

File: src/config/ConfigDescriptions.hpp
Comment (vaxerski) [reply src/config/ConfigDescriptions.hpp:460]: CHOICE cannot have SRangeData, it has to have SChoiceData

Comment (vaxerski) [reply src/config/ConfigDescriptions.hpp:460]: yes, looks fine.

these are descriptions that can be queried by `hyprctl descriptions` and used by settings daemons / GUIs

**Review by vaxerski — DISMISSED** [review]
thanks for the patience!

**Comment by vaxerski** [comment]
clang-format needed

**Review by vaxerski — APPROVED** [review]

**Pull request merged**
